### PR TITLE
plugins/cmp-tabby: refactor

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

**Comment by MattSturgeon** [comment]
Do you think this would be more maintainable as its own `plugins/by-name` module?

It can still be a cmp plugin in the same way that (e.g.) otter does it.

**Comment by MattSturgeon** [comment]
This can be rebased without bf6a68f406577718d31c67103769b7816e46b420, right?

File: plugins/by-name/cmp-tabby/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "cmp-tabby";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  imports = [
+    { cmpSourcePlugins.cmp_tabby = "cmp-tabby"; }
```
Comment (MattSturgeon) [inline plugins/by-name/cmp-tabby/default.nix:12]: IDK why, I need to double check this every time...

It is correct though :+1: 

https://github.com/nix-community/nixvim/blob/5abe382c54eff3cd3977733e9d268ab8994f397b/plugins/cmp/auto-enable.nix#L38

**Review by MattSturgeon — APPROVED** [review]
Awesome

**Comment by GaetanLepage** [comment]
@mergify queue

**Pull request merged**